streams/src/test/java/org/apache/kafka/streams/processor/internals/StandbyTaskTest.java

```diff
@@ -177,10 +181,7 @@ public void shouldThrowLockExceptionIfFailedToLockStateDirectory() throws IOExce
 
     @Test
     public void shouldTransitToRunningAfterInitialization() {
-        EasyMock.expect(stateManager.changelogOffsets()).andStubReturn(Collections.emptyMap());
-        stateManager.registerStateStores(EasyMock.anyObject(), EasyMock.anyObject());
-        EasyMock.expect(stateManager.changelogPartitions()).andReturn(Collections.emptySet()).anyTimes();
-        EasyMock.replay(stateManager);
+        stateManager.registerStateStores(any(), any());
```

Comment:
This is also a call on the mock which makes the verification on line 199 always true, independently of the call under test.

Reply:
Done.

Reply:
I meant that you should remove the call on the mock, because mocks should not be directly called in the test code. Mocks should be called within the call under test. In this case this is `task.initializeIfNeeded()`. Here it is really important that `stateManager.registerStateStores()` is only called once, since a second call to `task.initializeIfNeeded()` should not register the state stores again. The test should be like this:
```
    public void shouldTransitToRunningAfterInitialization() {
        task = createStandbyTask();

        assertEquals(CREATED, task.state());

        task.initializeIfNeeded();

        assertEquals(RUNNING, task.state());

        // initialize should be idempotent
        task.initializeIfNeeded();

        assertEquals(RUNNING, task.state());
        verify(stateManager).registerStateStores(any(), any());
    }
```
You are right to say that the stubbings were unnecessary here since they returned what is the default in Mockito.